File: api/src/main/java/com/fundmate/api/service/CategoryService.java
package com.fundmate.api.service;

import com.fundmate.api.dto.request.CategoryRequest;
import com.fundmate.api.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {
    CategoryResponse createCategory(CategoryRequest categoryRequest);
    List<CategoryResponse> getAllCategories();
    CategoryResponse getCategoryById(Long id);
    void deleteCategory(Long id);
}

File: frontend/src/services/category.service.js
import api from "./api";

export const categoryService = {
  async getCategories() {
    const response = await api.get("/categories");
    return response.data;
  },

  async createCategory(categoryData) {
    const response = await api.post("/categories", categoryData);
    return response.data;
  },

  async deleteCategory(id) {
    const response = await api.delete(`/categories/${id}`);
    return response.data;
  },
};
